01/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+)
+
+func main() {
+	// open the file
+	file, err := os.Open("input")
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	// read line by line
+	scan := bufio.NewScanner(file)
+	depth := 0
+	counter := 0
+	for scan.Scan() {
+		newDepth, err := strconv.Atoi(scan.Text())
+		if err != nil {
+			fmt.Printf("Error converting text to int. %s\n", err)
+		}
+
+		if depth == 0 {
+			fmt.Printf("%d (N/A - no previous measurement)\n", newDepth)
+		} else {
```

Comment:
Can you think of a way to write this code without the else statement?

Not an error, but the more indentation the harder to read, as a rule of thumb I try to stay away from else statements as much as possible.

Reply:
it will get simplified in the next commit